panther_hardware_interfaces/src/panther_imu.cpp

```diff
@@ -0,0 +1,513 @@
+{
+  const auto cc_mag_field = info_.hardware_parameters.at("cc_mag_field");
+  const auto cc_offset0 = info_.hardware_parameters.at("cc_offset0");
+  const auto cc_offset1 = info_.hardware_parameters.at("cc_offset1");
+  const auto cc_offset2 = info_.hardware_parameters.at("cc_offset2");
+  const auto cc_gain0 = info_.hardware_parameters.at("cc_gain0");
+  const auto cc_gain1 = info_.hardware_parameters.at("cc_gain1");
+  const auto cc_gain2 = info_.hardware_parameters.at("cc_gain2");
+  const auto cc_t0 = info_.hardware_parameters.at("cc_t0");
+  const auto cc_t1 = info_.hardware_parameters.at("cc_t1");
+  const auto cc_t2 = info_.hardware_parameters.at("cc_t2");
+  const auto cc_t3 = info_.hardware_parameters.at("cc_t3");
+  const auto cc_t4 = info_.hardware_parameters.at("cc_t4");
+  const auto cc_t5 = info_.hardware_parameters.at("cc_t5");
+
+  params_.cc_mag_field = hardware_interface::stod(cc_mag_field);
+  params_.cc_offset0 = hardware_interface::stod(cc_offset0);
+  params_.cc_offset1 = hardware_interface::stod(cc_offset1);
+  params_.cc_offset2 = hardware_interface::stod(cc_offset2);
+  params_.cc_gain0 = hardware_interface::stod(cc_gain0);
+  params_.cc_gain1 = hardware_interface::stod(cc_gain1);
+  params_.cc_gain2 = hardware_interface::stod(cc_gain2);
+  params_.cc_t0 = hardware_interface::stod(cc_t0);
+  params_.cc_t1 = hardware_interface::stod(cc_t1);
+  params_.cc_t2 = hardware_interface::stod(cc_t2);
+  params_.cc_t3 = hardware_interface::stod(cc_t3);
+  params_.cc_t4 = hardware_interface::stod(cc_t4);
+  params_.cc_t5 = hardware_interface::stod(cc_t5);
+}
+
+void PantherImuSensor::ReadMadgwickFilterParams()
+{
+  const auto gain = info_.hardware_parameters.at("gain");
+  const auto zeta = info_.hardware_parameters.at("zeta");
+  const auto mag_bias_x = info_.hardware_parameters.at("mag_bias_x");
+  const auto mag_bias_y = info_.hardware_parameters.at("mag_bias_y");
+  const auto mag_bias_z = info_.hardware_parameters.at("mag_bias_z");
+  const auto use_mag = info_.hardware_parameters.at("use_mag");
+  const auto stateless = info_.hardware_parameters.at("stateless");
+  const auto remove_gravity_vector = info_.hardware_parameters.at("remove_gravity_vector");
+
+  CheckMadgwickFilterWorldFrameParam();
+
+  params_.gain = hardware_interface::stod(gain);
+  params_.zeta = hardware_interface::stod(zeta);
+  params_.mag_bias_x = hardware_interface::stod(mag_bias_x);
+  params_.mag_bias_y = hardware_interface::stod(mag_bias_y);
+  params_.mag_bias_z = hardware_interface::stod(mag_bias_z);
+  params_.use_mag = hardware_interface::parse_bool(use_mag);
+  params_.stateless = hardware_interface::parse_bool(stateless);
+  params_.remove_gravity_vector = hardware_interface::parse_bool(remove_gravity_vector);
+}
+
+void PantherImuSensor::CheckMadgwickFilterWorldFrameParam()
+{
+  const auto world_frame = info_.hardware_parameters.at("world_frame");
+
+  auto it = kImuWorldFramesMap.find(world_frame);
```

Comment:
Using a map for this class member seems like overkill in my opinion.

I would even suggest a simpler approach such as:
```c++
if (world_frame == "ned") world_frame_  = WorldFrame::NED;
if (world_frame == "nwu") world_frame_  = WorldFrame::NWU;
if (world_frame == "enu") world_frame_  = WorldFrame::ENU;
```